### DISCUSSION TITLE
Snapshot missing on Livewire component with id: P58PshGixUDNPtlAkLaq

### DISCUSSION THREAD

**Discussion by mikebronner** [livewire/livewire#7952] [1. Help]
I am getting this console error in the following scenario:

1. Parent component "Customer" has child component "Form".
2. The form is used to update the customer information and save to the database using the Customer model.
3. On save the model broadcasts an even using Echo, for which the Customer component is listening.
4. The Customer component refreshes when receiving the broadcast message.
5. The original Form component now receives the livewire response from the server, but since the Customer component refreshed, the original Form component has been refreshed as well and no longer exists, throwing the following error:
  ```
  [Error] Snapshot missing on Livewire component with id: P58PshGixUDNPtlAkLaq
	Component (livewire.js:4400)
	initComponent (livewire.js:4491)
	(anonymous function) (livewire.js:8383)
	(anonymous function) (livewire.js:1887)
	(anonymous function) (livewire.js:916)
	forEach
	(anonymous function) (livewire.js:916)
	walk (livewire.js:839)
	walk (livewire.js:844)
	walk (livewire.js:844)
	walk (livewire.js:844)
	walk (livewire.js:844)
	(anonymous function) (livewire.js:914)
	deferHandlingDirectives (livewire.js:1397)
	initTree (livewire.js:913)
	(anonymous function) (livewire.js:861)
	(anonymous function) (livewire.js:1074)
	forEach
	onMutate (livewire.js:1074)
  ```

Any thoughts on how to avoid this scenario?

Here are a few caveats:
- the form component is a child component because it pertains to the same customer.
- the Customer component has a button that makes the Form component visible.

**Comment by timetorock** [comment]
Hi, any success with this? Have similar problems, where child component updates, and after that send a refresh event to parent component. For some reason after this, child component miss snapshot and next action which will add new child row, will trigger snapshot error for previously used component.

**Reply by mikebronner** [reply]
(Unfortunately not yet ... I am no longer active on that project, so I don't know if the issue has been fixed by upgrading to Livewire v3.)

**Reply by PhiloNL** [reply]
Please check if you have an unclosed/trailing element, for example, you might have a missing `</div>` tag. This will often result in this error.
